scripts/apps/master-desk/components/FilterPanelComponent.tsx

```diff
@@ -71,7 +93,32 @@ export class FilterPanelComponent extends React.Component<IProps, IState> {
         this.props.onDeskFilterChange('');
     }
 
+    clearFilters() {
+        const filter = {
+            slugline: [],
+            headline: [],
+            byline: [],
+        };
+
+        this.setState({filter: filter});
+
+        this.props.onFilterChange({});
+    }
+
     render() {
+        const isDisplayClearFilters = () => {
```

Comment:
It can be written in a shorter form:

```
const showClearFiltersButton = Object.keys(this.props.filters).some((key) => this.props.filters[key].length !== 0);
```

Regarding performance, it wouldn't have an impact here, but it's a good practice to stop iterating as soon as you find the value that you are looking for. When you set `isDisplay = true;` on line 115, there is no point in continuing the loop anymore because you already have the result.